### PR TITLE
[ETHEREUM-CONTRACTS] GDA View Function Fixes

### PR DESCRIPTION
**Pull request by hellwolf**
## Notes

1. How to re-build /settled amount/ in subgraph for GDA?
2. Missing GDA View Functions:
  - ~~GDA Flow Info (IGDAv1 & SuperTokenV1Library)~~
  - ~~getNetFlow misses the distribution flow rate~~
  - ~~libtoken.getBufferAmountByFlowRate needs a comment; or better, share the impl between CFA & GDA using a free-ranged function.~~

Other notes:

- getBufferAmountByFlowRate 

--------------
Code review todos
1. Missing functions in IGDAv1
  - [x] getFlow
  - [x] getAccountFlowInfo
2. Missing functions in SuperTokenV1Library
  - [x] getGDAFlowInfo
  - [x] getNetFlowInfo, getCFANetFlowInfo, getGDANetFlowInfo
  - [x] getPoolAdjustmentFlowRate
3. New function in ISuperfluidPool
  - [x] getTotalAmountReceivedByMember, need its assertions in some test cases.
  - [x] remove getMemberData
  - [x] rename getIndex -> poolOperatorGetIndex()
  - [x] Make interface files minimum

### CODE REVIEW

File: packages/sdk-redux/src/reduxSlices/rtkQuery/cacheTags/invalidateSpecificCacheTagsForEvents.ts [hunk 1]
```diff
@@ -153,9 +153,8 @@ const getEventSpecificTags = (event: AllEvents, chainId: number) => {
             return [];
         default:
             console.warn(
-                `Unknown event [${
-                    (event as any)?.name
-                }] in "invalidateCacheTagsForEvents" for @superfluid-finance/sdk-redux. Cache might not be invalidated properly.`
+                `Unknown event [${(event as any)
```
Comment (hellwolf) [inline packages/sdk-redux/src/reduxSlices/rtkQuery/cacheTags/invalidateSpecificCacheTagsForEvents.ts]: how did it creep into this PR? what made this branch different than the dev branch.

**Comment by kasparkallas** [comment]
Can I get a script kiddie explanation on how I can use the view functions to fetch how much has a single pool member received from a GDA pool?

File: packages/sdk-redux/src/reduxSlices/rtkQuery/cacheTags/invalidateSpecificCacheTagsForEvents.ts (hunk 1 above)
Comment (0xdavinchee) [reply packages/sdk-redux/src/reduxSlices/rtkQuery/cacheTags/invalidateSpecificCacheTagsForEvents.ts]: didn't yarn install, it's resolved now

**Comment by kasparkallas** [comment]
> Can I get a script kiddie explanation on how I can use the view functions to fetch how much has a single pool member received from a GDA pool?

@0xdavinchee added the function `getTotalAmountReceivedByMember`

File: packages/ethereum-contracts/contracts/agreements/gdav1/SuperfluidPool.sol [hunk 2]
```diff
@@ -214,6 +242,26 @@ contract SuperfluidPool is ISuperfluidPool, BeaconProxiable {
         return Value.unwrap(PDPoolMemberMU(pdPoolIndex, pdPoolMember).rtb(Time.wrap(time)));
     }
 
+    /// @inheritdoc ISuperfluidPool
+    function getMemberData(address memberAddr) external view override returns (MemberData memory memberData) {
+        return _membersData[memberAddr];
+    }
+
+    /// @inheritdoc ISuperfluidPool
+    function getTotalAmountReceivedByMember(address memberAddr) external view override returns (uint256) {
```
Comment (hellwolf) [inline packages/ethereum-contracts/contracts/agreements/gdav1/SuperfluidPool.sol:265]: add the assertion of `getTotalAmountReceivedByMember` usage in some of the existing test cases.

**Comment by hellwolf** [comment]
```
1. Missing functions in IGDAv1
  - [x] getFlow
  - [x] getAccountFlowInfo
2. Missing functions in SuperTokenV1Library
  - [x] getGDAFlowInfo
  - [-] getNetFlowInfo, getCFANetFlowInfo, getGDANetFlowInfo
  - [ ] getPoolAdjustmentFlowRate
3. New function in ISuperfluidPool
  - [-] getTotalAmountReceivedByMember, need its assertions in some test cases.
  - [ ] remove getMemberData
  - [ ] rename getIndex -> poolOperatorGetIndex()
  - [ ] Make interface files minimum
```

**Comment by hellwolf** [comment]
Update change log, then we are done.

**Comment by hellwolf** [comment]
Let's merge! Please approve, instead.

**Review by 0xdavinchee — APPROVED** [review]
LGTM

File: packages/ethereum-contracts/contracts/interfaces/agreements/gdav1/ISuperfluidPool.sol [hunk 3]
```diff
@@ -83,9 +65,14 @@ interface ISuperfluidPool is IERC20 {
     /// @param time The time to query
     function getDisconnectedBalance(uint32 time) external view returns (int256 balance);
 
+    /// @notice The total amount received by `memberAddr` in the pool
```
Comment (kasparkallas) [inline packages/ethereum-contracts/contracts/interfaces/agreements/gdav1/ISuperfluidPool.sol:68]: Should make it more clear whether this includes both the connected and unconnected parts

**Pull request merged**